src/macaron/slsa_analyzer/git_service/gitlab.py

```diff
@@ -117,7 +123,74 @@ def clone_repo(self, clone_dir: str, url: str) -> None:
             If there is an error cloning the repository.
         """
         clone_url = self.construct_clone_url(url)
-        git_url.clone_remote_repo(clone_dir, clone_url)
+        repo = git_url.clone_remote_repo(clone_dir, clone_url)
+
+        # If ``git_url.clone_remote_repo`` return an Repo instance, this means that the repository is freshly cloned
+        # with the token embedded URL. We will set its value back to the original non-token URL.
+        # If ``git_url.clone_remote_repo`` returns None, it means that the repository already exists so we don't need
+        # to do anything.
+        if repo:
+            try:
+                origin_remote = repo.remote("origin")
+            except ValueError as error:
+                raise CloneError("Cannot find the remote origin for this repository.") from error
+
+            origin_remote.set_url(url)
+
+    def check_out_repo(self, git_obj: Git, branch: str, digest: str, offline_mode: bool) -> Git:
+        """Checkout the branch and commit specified by the user of a repository.
+
+        For GitLab, this method set the origin remote URL of the target repository to the token-embedded URL if
+        a token is available before performing the checkout operation.
+
+        After the checkout operation finishes, the origin remote URL is set back again to ensure that no token-embedded
+        URL remains.
+
+        Parameters
+        ----------
+        git_obj : Git
+            The Git object for the repository to check out.
+        branch : str
+            The branch to check out.
+        digest : str
+            The sha of the commit to check out.
+        offline_mode: bool
+            If true, no fetching is performed.
+
+        Returns
+        -------
+        Git
+            The same Git object from the input.
+
+        Raises
+        ------
+        RepoError
+            If there is error while checkout the specific branch and digest.
+        """
+        remote_origin_url = git_url.get_remote_origin_of_local_repo(git_obj)
+
+        try:
+            origin_remote = git_obj.repo.remote("origin")
+        except ValueError as error:
+            raise RepoError("Cannot find the remote origin for this repository.") from error
+
+        try:
+            reconstructed_url = self.construct_clone_url(remote_origin_url)
+        except CloneError as error:
+            raise RepoError("Internal error prevent preparing the repo for the analysis.") from error
+
+        origin_remote.set_url(reconstructed_url, remote_origin_url)
+
+        check_out_status = git_url.check_out_repo_target(git_obj, branch, digest, offline_mode)
+
+        origin_remote.set_url(remote_origin_url, reconstructed_url)
+
+        if not check_out_status:
+            raise RepoError(
+                f"Internal error when checking out branch {branch} and commit {digest} for repo {git_obj.project_name}."
```

Comment:
Same comment as above.

Reply:
Fixed in https://github.com/oracle/macaron/pull/354/commits/77e4d140d15d418ad19099ba92fccea67262f851